File: src/main/java/com/rabit/api/dto/response/LoginResponse.java
package com.rabit.api.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LoginResponse {

    private UUID userId;

    private String username;

    private String email;

    private String firstName;

    private String lastName;

    private String token;

    private long expiresIn;
}

File: dtos/responses/LoginResponse.js
class LoginResponse {
  constructor(user, token, expiresIn) {
    this.userId = user.userId;
    this.username = user.username;
    this.email = user.email;
    this.firstName = user.firstName || null;
    this.lastName = user.lastName || null;
    this.token = token;
    this.expiresIn = expiresIn;
  }
}
module.exports = LoginResponse;
